### DISCUSSION TITLE
Firebase Static Hosting Issue

### DISCUSSION THREAD

**Discussion by sanjaiyan-dev** [vercel/next.js#30533] [Help] [answered]
Hi,

I have created website using _Next js_ (`next export`) and deployed on _Firebase_ but I am facing few issues 😞 

**With SPA option enabled through Firebase CLI**
 

- Users are getting `/` page even they are made initial request to other routes.

> eg-: `/docs` -> `/` <br/>  `/contact` -> `/` <br/> This issue is happening only in _first/initial_ request only

**Without SPA options enabled through Firebase CLI**

- Users are getting 404 page even though they are made their initial valid request :(

> eg-: `/docs` -> 404 Page <br/>  `/contact` -> 404 Page <br/> This issue is happening only in _first/initial_ request only
---
_Thanks in advance & Extremely Sorry if I made any mistakes_

_Originally posted by @SanjaiyanUnofficial in https://github.com/vercel/next.js/discussions/30197_

**Comment by DevAtHeart** [comment, marked as answer]
Try adding this to the firebase.json file under "hosting":
`"cleanUrls": true"`

**Reply by sanjaiyan-dev** [reply]
Thanks a lot :)
